commit 91152a049546b75e5b0ce334198a9231a1e7d15a
Author: David Faure <faure@kde.org>
Date:   Mon Oct 3 23:07:05 2016 +0200

    Don't remove tasks from the inbox when associating them to a context.
    
    Summary:
    Otherwise the user has to go through all projects *and* all contexts
    for the daily task review. With this change, going through all projects
    (and the inbox, to sort tasks into projects) is enough.
    
    The fix involves a lot more kiwis :-)
    
    Test Plan: make test + manual testing
    
    Reviewers: ervin, franckarrecot
    
    Reviewed By: franckarrecot
    
    Subscribers: franckarrecot, #zanshin
    
    Differential Revision: https://phabricator.kde.org/D2923

diff --git a/src/akonadi/akonaditaskqueries.cpp b/src/akonadi/akonaditaskqueries.cpp
index 9909622d..79defcf6 100644
--- a/src/akonadi/akonaditaskqueries.cpp
+++ b/src/akonadi/akonaditaskqueries.cpp
@@ -93,8 +93,7 @@ TaskQueries::TaskResult::Ptr TaskQueries::findInboxTopLevel() const
     auto fetch = m_helpers->fetchItems(StorageInterface::Tasks);
     auto predicate = [this] (const Akonadi::Item &item) {
         const bool excluded = !m_serializer->isTaskItem(item)
-                           || !m_serializer->relatedUidFromItem(item).isEmpty()
-                           || m_serializer->hasContextTags(item);
+                           || !m_serializer->relatedUidFromItem(item).isEmpty();
 
         return !excluded;
     };
